openshift-kube-apiserver/admission/customresourcevalidation/node/validate_node_config.go

```diff
@@ -103,11 +121,26 @@ func (configNodeV1) ValidateUpdate(_ context.Context, uncastObj runtime.Object,
 	if err := validateConfigNodeForExtremeLatencyProfile(obj, oldObj); err != nil {
 		allErrs = append(allErrs, err)
 	}
+	if err := c.validateMinimumKubeletVersion(obj); err != nil {
+		allErrs = append(allErrs, err)
+	}
 
 	return allErrs
 }
+func (c *configNodeV1) validateMinimumKubeletVersion(obj *configv1.Node) *field.Error {
+	fieldPath := field.NewPath("spec", "minimumKubeletVersion")
+	nodes, err := c.nodeLister().List(labels.Everything())
```

Comment:
We definitely need to make sure the informer is synced before using it here, right? IIUC, if a configv1 node update request can reach an apiserver before it syncs the informer, any parseable semver will be considered valid.

Reply:
this is not an informer but a lister, which I chose based on other admission plugins. I don't believe it has a method HasSynced or similar, though please correct me if I am not seeing something

Reply:
The lister retrieves values from a store managed by an informer. We need to know that that informer has synced, otherwise the lister is going to appear empty (https://github.com/kubernetes/sample-controller/blob/master/docs/controller-client-go.md#client-go-under-the-hood).

Reply:
added!